citesphere/src/main/java/edu/asu/diging/citesphere/web/user/AsyncCitationManager.java

```diff
@@ -86,8 +94,59 @@ public AsyncUpdateCitationsResponse getUpdateCitationsResponse(String taskId)
      * 
      * @param taskId id of the task that is not needed in memory
      */
-    public void clearTask(String taskId) {
-        taskTracker.remove(taskId);
+    public void clearUpdateTask(String taskId) {
+        updateTaskTracker.remove(taskId);
+    }
+
+    /**
+     * Deletes Citations asynchronously
+     * 
+     * @param user           User accessing Zotero
+     * @param groupId        GroupId of the Citations
+     * @param citationIdList List of the Citations to be deleted
+     * @return AsyncDeleteCitationsResponse containing taskId and its current status
+     */
+    public AsyncDeleteCitationsResponse deleteCitations(IUser user, String groupId, List<String> citationIdList)
+            throws ZoteroConnectionException, ZoteroHttpStatusException {
+        String taskId = UUID.randomUUID().toString();
+        Future<Map<ItemDeletionResponse, List<String>>> futureTask = asyncDeleteCitationsProcessor.deleteCitations(user,
+                groupId, citationIdList);
+        deleteTaskTracker.put(taskId, futureTask);
+        AsyncDeleteCitationsResponse asyncResponse = new AsyncDeleteCitationsResponse();
+        asyncResponse.setTaskID(taskId);
+        asyncResponse.setTaskStatus(AsyncTaskStatus.PENDING);
+        return asyncResponse;
+    }
+
+    /**
+     * Gets the current status of the deletion task. If the task is completed, it
+     * returns the status for each Citation which was requested to be deleted.
+     * 
+     * @param taskId Id of the task
+     * @return AsyncDeleteCitationsResponse containing task id, task status, and
+     *         response if the task is completed
+     */
+    public AsyncDeleteCitationsResponse getDeleteCitationsResponse(String taskId)
+            throws ExecutionException, InterruptedException {
+        AsyncDeleteCitationsResponse response = new AsyncDeleteCitationsResponse();
+        Future<Map<ItemDeletionResponse, List<String>>> futureTask = deleteTaskTracker.get(taskId);
+        response.setTaskID(taskId);
+        if (futureTask.isDone()) {
+            response.setTaskStatus(AsyncTaskStatus.COMPLETE);
+            response.setResponse(futureTask.get());
+        } else {
+            response.setTaskStatus(AsyncTaskStatus.PENDING);
+        }
+        return response;
+    }
+
+    /**
+     * Removes the Citation deletion task from memory
+     * 
+     * @param taskId Id of the task
+     */
+    public void clearDeleteTask(String taskId) throws InterruptedException, ExecutionException {
```

Comment:
I'm going back and forth on this, but I think the better approach is to have a cron job that checks the age of tasks and removes them (say after 24 hours). Otherwise, if the application isn't restarted for a long time and many people just close the browser or navigate away from the page the tasks are never cleared and pile up. If you just add a time stamp and use Spring's cron job feature (might be a new plugin, not sure), it should be fairly straight forward and you can remove the frontend logic to remove tasks.

Reply:
Okay. I will take a look.